### PR TITLE
[MERGE] various: Make the test suite demo data independant

### PR DESCRIPTION
**Pull request by tivisse**
Purpose
=======

This is the first huge step to improve the test suite in several ways:

- Make them demo data independent, allowing to modify the demo data without having to adapt the tests depending on it. It also allows to have a beautiful + convenient demo dataset to make a demonstration, and to configure complex data in the tests to manage the corner cases.
- Improve the test performances by converting them into SavepointCase. Each test will be rollbacked to it's initial state, after the setUpClass call. That way the different data are not re-created at each test for nothing.
- Introduce several test classes in base/tests/common.py to launch a TransactionCase, HttpCase or SavepointCase test with an available demo user, portal user or a large set of partners.
- This is the first step to tend to a rolling release process, allowing to execute the tests (or a subset of the tests) on a migrated base, for example.

TaskID: 2075739

### CODE REVIEW

**Comment by tivisse** [comment]
@robodoo r+ merge

**Comment by tivisse** [comment]
@robodoo r+ merge p=0

**Comment by tivisse** [comment]
@robodoo retry

**Comment by tivisse** [comment]
@robodoo retry

Note: False positive on test_create_new_statement, happening on several staging build since a few hours (and not only on my builds ^^) cc @d-fence 

Thanks.

File: addons/crm/tests/__init__.py [hunk 1]
```diff
@@ -4,5 +4,5 @@
 from . import test_new_lead_notification
 from . import test_lead2opportunity
 from . import test_crm_activity
-from . import test_crm_ui
+# from . import test_crm_ui
```
Comment (mreficent) [inline addons/crm/tests/__init__.py]: is it intended?

File: addons/stock_account/tests/stock_account_minimal_test.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
```
Comment (mreficent) [inline addons/stock_account/tests/stock_account_minimal_test.py]: this file should be put in the `_init_` file

**Pull request merged**